### DISCUSSION TITLE
ESP32 S3 Wroom 1 DevKit C N16R8

### DISCUSSION THREAD

**Discussion by Waz-Cpt** [revk/xESP32-Faikout#502] [General]
Sorry to jump on the band wagon here and hijack a discussion. I am quite a novice when it comes to understand the compiling process of the ESP IDF. I have only worked with ESPhome and with Arduino in PlatformIO.

So my question is what are the steps to follow to build this project for a different board type. I have a full size ESP32 S3 WROOM 1 N16R8.

So I know the SoC should work with the standard build but I am worried that the different RAM and Storage might cause issues.

Any advice would be really appreciated.

I have attempted to flash the N4R2 bins to the module and it did not like the bootloader file.

**Comment by revk** [comment]
The Makefile and scripts handle a variety of builds, but I have not created partition files or builds for N16 (did not know there was an N16). I would expect the N4R2 to work though.

**Reply by Waz-Cpt** [reply]
Maybe there is something I am missing while attempting to flash. Your instructions mention using 0x0 for the bootloader bin.

But the ESP Tool says it does not recognise the file as a bootloader file. Do I need to fully erase the flash considering that I had an ESPhome project previously on the device?

**Reply by revk** [reply]
Probably, you can't use my boot loader with your code. There is a web based WLED flasher, that should work. See link on https://hiwtsi.uk/led.cgi

**Reply by Waz-Cpt** [reply]
I am attempting to flash your boot loader and firmware to test out the control of one of our Daikin aircons. Your bin file results in a bootloop on my ESP32 S3.

**Reply by revk** [reply]
Ah, I see, Ok there is a link with the command for that, surprised if that is not working.

**Reply by Waz-Cpt** [reply]
I was attempting to use the [https://espressif.github.io/esptool-js/](url) web tool to flash the ESP with your BIN files.

I have attempted both the N4R2 and N8 firmware and both result in the same thing.

ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x3 (RTC_SW_SYS_RST),boot:0x28 (SPI_FAST_FLASH_BOOT)
Saved PC:0x403cd8cd
SPIWP:0xee
mode:DIO, clock div:2
load:0x3fce3810,len:0x48c
load:0x403c9700,len:0x4
load:0x403c9704,len:0xa4c
load:0x403cc700,len:0x26e4
entry 0x403c986c

This is repeated over and over on the serial debug.

What link are you referring to?

**Reply by revk** [reply]
https://github.com/revk/ESP32-Faikin/blob/main/Manuals/Setup.md
<img width="1056" alt="Screenshot 2024-09-26 at 10 05 52" src="https://github.com/user-attachments/assets/9e199502-9fbc-45c7-9da7-b52ee1bdf6e3">

**Reply by Waz-Cpt** [reply]
Ok, I missed this. I have followed this information and now have a different boot loop error which is promising.

![image](https://github.com/user-attachments/assets/d7da8e28-49f4-44f0-bcf6-41d024d60a7c)

**Reply by revk** [reply]
That sounds like the PSRAM is not the same settings.

Maybe try the `Faikin-S3-MINI-N8.bin` which does not try to use the PSRAM

**Reply by Waz-Cpt** [reply]
Ok, will try that. Do I then ignore the bootloader? What about the partition table and ota portion?

I have already tried flashing the S3 N8 bin to the device, but did not do it at 0x10000, did it at 0x1000.

**Reply by revk** [reply]
You need to flash it all, but does not matter if a smaller partition table

**Reply by Waz-Cpt** [reply]
What boot loader do I use for the N8?

Do I use all the same positioning for the bin files?

**Reply by revk** [reply]
Ah, I have added `Faikin-S3-MINI-N8-bootloader.bin` to `release` now

**Reply by Waz-Cpt** [reply]
Thank you. Will test that out.

**Reply by Waz-Cpt** [reply]
Bingo, successufullly flashed and can see a request for AP. I will connect it up to the aircon shortly and test if it works why my model.

Do you know if I need the 5v reference from the X50 connector or can I use the 5v from the voltage regulator?

**Reply by revk** [reply]
Usually not needed

**Reply by Waz-Cpt** [reply]
Just one strange thing, it will not let me capture the Wifi SSID or password. Is this restriction while it is not connected to the AC?

**Reply by revk** [reply]
Not sure what you mean, if not on WiFi it should come up as an AP and allow you to set WiFi. I don't think connected to AC changes that.

**Reply by Waz-Cpt** [reply]
What I mean is that I cannot capture the settings when I connect to the hotspot.

Will not let me type in any values for any of the fields. Tried from IPhone and from Windows PC with chrome.

**Reply by revk** [reply]
Ok that makes no sense. They are just html input fields!!!

**Reply by Waz-Cpt** [reply]
Tried erasing and then flashing again. Same result but the Free mem value is different.
Is it possible that this has something to do with storage and writable partition space?

**Reply by revk** [reply]
It normally works, maybe look at the HTML, but no way it should be able to stop you typing in an input box!

**Reply by Waz-Cpt** [reply]
What should I look at with the developer panel?

**Reply by revk** [reply]
I don't know, is it just an <input> field.
If so, you can type in to it, surely?

**Reply by Waz-Cpt** [reply]
It will not let me type in any of the fields.

**Reply by Waz-Cpt** [reply]
The only thing it lets me do other than pressing the save but, is remove the text from the timezone field.

Should I try the beta release?

**Reply by revk** [reply]
OK you have a browser and a page with <input> fields, and you cannot type. At this point I am not sure what to say other than "I did not supply/write the browser".

**Reply by Waz-Cpt** [reply]
I have tried from my phone with both chrome and safari, I have tried on my computer with both Edge and Chrome. It is all very strange.

**Reply by revk** [reply]
I agree, but if the HTML is OK and has an `<input>`, it is not me doing it wrong, is it?

**Reply by Waz-Cpt** [reply]
I think this is likely due to the PSRAM not being initiated and causing a block on any settings changes.

**Reply by revk** [reply]
It works with no PSRAM, that is not unusual.

But you say you cannot type in an input box, that is a browser issue.

**Reply by Waz-Cpt** [reply]
'<input` maxlength="0" size="-1" id="wifissid" name="_wifissid" onchange="this.name='wifissid';" value="" autocapitalize="off" autocomplete="off" spellcheck="false" autocorrect="off" `placeholder="">`
This is the html element that is coming through. Are you certain that using 0 as the maxlength is correct. Surely you should not be setting maxlength of 0 and some actual value.

**Reply by revk** [reply]
OK finally, a `maxlength` of 0 is wrong as is a `size` of `-1`

Try the beta release.

**Reply by Waz-Cpt** [reply]
Bingo, now I can input data.

**Reply by revk** [reply]
I've made it a release now, I did not realise that issue was in current release.
